packages/vscode-extension/src/utilities/packageManager.ts

```diff
@@ -1,15 +1,36 @@
 import path from "path";
-import { promises as fs } from "fs";
+import fs from "fs";
 import { command } from "./subprocess";
 import { getAppRootFolder } from "./extensionContext";
 import { isWorkspaceRoot } from "./common";
 import { Logger } from "../Logger";
+import { getLaunchConfiguration } from "./launchConfiguration";
+import { requireNoCache } from "./requireNoCache";
 
 export type PackageManagerInfo = {
   name: "npm" | "pnpm" | "yarn" | "bun";
   workspacePath?: string;
 };
 
+function isPackageManager(candidate: string): boolean {
+  const packageManagers = ["npm", "yarn", "pnpm", "bun"];
+  return packageManagers.includes(candidate);
+}
+
+async function getDirFilesSortedByModificationDate(dir: string) {
+  const files = await fs.promises.readdir(dir);
+
+  return files
+    .map((fileName) => ({
+      name: fileName,
+      time: fs.statSync(`${dir}/${fileName}`).mtime.getTime(),
+    }))
+    .sort((a, b) => a.time - b.time)
```

Comment:
this sorts in ascending order. I think we want to select one that's latest modification date not the one with the oldest